src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,42 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
-}
+  // write your code here0
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 
+  hide() {
+    this.hidden = true;
+  }
+}
 class Carnivore extends Animal {
   // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  bite(enemy) {
+    if (enemy instanceof Herbivore && enemy.hidden === false) {
+      enemy.health -= 50;
+
+      Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
```

Comment:
[CODE PERFOMANCE] - You're currently filtering the Animal.alive array every time the `bite` method is called. This can be inefficient if the array is large. Instead, you could check if the health of the `enemy` animal is less than or equal to 0 after it has been bitten, and if so, remove it from the array. This way, you're only filtering the array when necessary.